src/models/filter/filter-model.ts

```diff
@@ -0,0 +1,76 @@
+import { FilterConfig } from "./filter-config-item";
+import {BaseModel} from "../base-model";
+
+export enum FilterModelProperty {
+    min = 'min',
+    max = 'max',
+    selectMin = 'selectMin',
+    selectMax = 'selectMax',
+}
+
+export class FilterModel<T> extends BaseModel{
+    private filteredData: T[];
+    private data: T[];
+    private filterModel: FilterConfig;
```

Comment:
Let's think about another name for this property because it seems a bit strage to me to have `filterModel` property in `FilterModel` class.